lib/tapioca/gemfile.rb

```diff
@@ -105,14 +105,13 @@ def load_dependencies
     sig { returns([T::Enumerable[Spec], T::Array[String]]) }
     def materialize_deps
       deps = definition.locked_gems.dependencies.values
-      missing_specs = T::Array[String].new
-      materialized_dependencies = if definition.resolve.method(:materialize).arity == 1 # Support bundler >= v2.2.25
-        md = definition.resolve.materialize(deps)
-        missing_spec_names = md.missing_specs.map(&:name)
-        missing_specs = T.cast(md.missing_specs.map { |spec| "#{spec.name} (#{spec.version})" }, T::Array[String])
-        md.to_a.reject { |spec| missing_spec_names.include?(spec.name) }
-      else
-        definition.resolve.materialize(deps, missing_specs)
+      materialized_dependencies = definition.resolve.materialize(deps)
+      missing_spec_names = materialized_dependencies.missing_specs.map(&:name)
```

Comment:
Nitpick: Can we make this a set, so that the `include?` operation below does not end up being quadratic:
```suggestion
      missing_spec_names = materialized_dependencies.missing_specs.map(&:name).to_set
```